### ISSUE TITLE
Add `aria-disabled` to `aria` property of button components

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#1789]
Scope
---
1. The first rule of ARIA is "If you can use a native HTML element or attribute with the semantics and behavior you require already built in, instead of repurposing an element and adding an ARIA role, state or property to make it accessible, then do so." `aria-disabled` is an **exception** to that rule. It is one of the only cases where an ARIA attribute may be better than the native disabled attribute HTML equivalent.

2. In `loading` state there is an accessibility contrast issue between text and background. Due that this state has not the `disabled` attribute, the contrast hast to be enhanced. Also check accessibility optimization by adding state property:
https://accessabilly.com/an-accessible-microinteractions-button/ 

Affected components:
- `Button`
- `Button-Pure`
- `P-Switch`

Notes  
---
- MDN Docs: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled 
- Slack Community Reference: https://porschedev.slack.com/archives/CT7AVHTTQ/p1646227046142409 

Acceptance criteria
---
- [x] Shadowed `button` of `Button`, `Button-Pure` and `P-Switch` is using `aria-disabled` instead of `disabled` when custom `disabled` prop on component is set.
- [x] In case `disabled` prop is set then click events are prevented (like already implemented for `loading` state)
- [x] Screen reader announces `loading` state in case `loading` prop is set to `true` (works with VoiceOver on macOS without any change)
- [x] Styling of disabled + loading state meets accessibility contrast requirements (not needed for a11y checker in case aria-disabled="true" is set)
- [x] Decision why `aria-disabled` is preferred over native `disabled` state is documented in storefront

Subtasks
---
- [x] Check issue mentioned in https://porschedev.slack.com/archives/CT7AVHTTQ/p1654583910048179
